### ISSUE TITLE
Scope indent check can go into infinite loop due to some parse errors

### ISSUE THREAD

**Issue by rasmusbe** [squizlabs/PHP_CodeSniffer#667]
This code gives an infinite loop in File.php and prints out this notices over and over again:

PHP Notice:  Undefined index:  in php_codesniffer/CodeSniffer/File.php on line 3479
PHP Notice:  Undefined index:  in php_codesniffer/CodeSniffer/File.php on line 3491

(its because of a ; instead of : in the end of the foreach, but it doesn't warn about it correctly)

```php
<?php for ( $year = $first_year; $year <= $current_year; $year++ ): ?>
	<?php foreach ( $events as $event ); ?>
		<li class="item item-press">
		<strong class="item-date"><?php echo get_the_date( 'j M', $event ); ?></strong>
		<a href="<?php echo get_permalink( $event );?>"><?php echo $event->post_title;?></a>
		<span class="item-city"><?php the_papi_field( 'papi_city' ); ?></span>
		</li>
	<?php endforeach; ?>
<?php endfor; ?>

**Comment by aik099** [comment]
Which PHP_CodeSniffer version are you using?

**Comment by gsherwood** [comment]
PHPCS doesn't check syntax unless you have a custom standard that includes a sniff to do it (there is one in the Generic standard) so you're not going to get syntax errors reported if you are using the built-in standards.

Fixing the parse error is the way to work around this of course, but I'll see if I can get PHPCS to stop throwing errors as well.

**Comment by rasmusbe** [comment]
@aik099 PHP_CodeSniffer version 2.3.3 (and using [Wordpress Coding Standards](https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards))
